### PR TITLE
Feature/#1854 create edit tile

### PR DESCRIPTION
**Pull request by aminbksi**

### CODE REVIEW

File: src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/RemoveTileDialog/RemoveTileDialog.tsx [hunk 1]
```diff
@@ -45,7 +43,7 @@ const RemoveTileDialog: FC = () => {
         TOAST_COMMON_OPTIONS
       );
     }
-  };
+  }, []);
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/Dashboard/components/RemoveTileDialog/RemoveTileDialog.tsx:46]: empty dependencies

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx [hunk 2]
```diff
@@ -1,62 +1,155 @@
...
+  createRequestPending,
+  updateRequestPending,
+  uploadRequestPending
 }) => {
   const theme = useTheme();
+  const {t} = useTranslation();
 
-  const handleImage = (file: File | undefined) => {
+  const [image, setImage] = useState<File>();
+  const [imageError, setImageError] = useState<boolean>(false);
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:39]: Can't you use `useForm`?

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx [hunk 3]
```diff
@@ -1,62 +1,155 @@
...
+
+  const handleSubmit = async () => {
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:41]: Maybe use `useCallback`?

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx [hunk 4]
```diff
@@ -1,62 +1,155 @@
...
+      ) : (
+        <>
+          <styled.FileUploaderItem>
+            <styled.TileImageUpload className={cn(imageError && 'error')}>
+              {(image || currentTile?.hash) && (
+                <styled.ImagePreview
+                  src={
+                    (image && URL.createObjectURL(image)) ||
+                    (currentTile?.hash &&
+                      `${appVariables.RENDER_SERVICE_URL}/get/${currentTile?.hash}`) ||
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:129]: I'm thinking, maybe we should move it to renderRepository and have an endpoint:

`get: '${appVariables.RENDER_SERVICE_URL}/get/:hash`

and then here use `generatePath(api.renderRepository.get, {hash: currentTile.hash}`?
Or something of this style

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx [hunk 5]
```diff
@@ -1,45 +1,139 @@
...
+    handleSubmit,
+    reset
+  } = useForm<VideoTileFormInterface>();
+
+  const formSubmitHandler: SubmitHandler<VideoTileFormInterface> = async (
+    data: VideoTileFormInterface
+  ) => {
+    if (!currentTile?.id) {
+      const isSucceed = await createTile(spaceId, data);
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx:47]: remove

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx [hunk 6]
```diff
@@ -1,45 +1,139 @@
...
+            isDanger
+            isCloseButton
+          />,
+          TOAST_COMMON_OPTIONS
+        );
+      }
+    } else {
+      const isSucceed = await updateTile(currentTile.id, data);
+      await fetchDashboard(spaceId);
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx:74]: Remove

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/TextTileForm/TextTileForm.tsx [hunk 7]
```diff
@@ -1,62 +1,156 @@
...
+    handleSubmit,
+    reset
+  } = useForm<TextTileFormInterface>();
+
+  const formSubmitHandler: SubmitHandler<TextTileFormInterface> = async (
+    data: TextTileFormInterface
+  ) => {
+    if (!currentTile?.id) {
+      const isSucceed = await createTile(spaceId, data);
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/TextTileForm/TextTileForm.tsx:46]: remove

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/TextTileForm/TextTileForm.tsx [hunk 8]
```diff
@@ -1,62 +1,156 @@
...
+            text={t('messages.tileCreateError')}
+            isDanger
+            isCloseButton
+          />,
+          TOAST_COMMON_OPTIONS
+        );
+      }
+    } else {
+      const isSucceed = await updateTile(currentTile.id, data);
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/TextTileForm/TextTileForm.tsx:72]: remove

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx [hunk 9]
```diff
@@ -1,62 +1,155 @@
...
+  const handleSubmit = async () => {
+    if (!currentTile?.id) {
+      if (!image) {
+        setImageError(true);
+        return;
+      }
+      const isSucceed = await createTile(spaceId, image);
+      setImageError(false);
+
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:50]: remove

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx [hunk 10]
```diff
@@ -1,62 +1,155 @@
...
+      }
+    } else {
+      if (!image) {
+        setImageError(true);
+        return;
+      }
+      const isSucceed = await updateTile(currentTile?.id, image);
+      setImageError(false);
+
+      // @ts-ignore
```
Comment (Daboo1998) [inline src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:82]: remove

File: src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts [hunk 11]
```diff
@@ -11,11 +22,97 @@ const DashboardStore = types.compose(
...
       tileList: types.optional(TileList, {}),
-      dashboard: types.optional(Dashboard, {}),
-      tileFormStore: types.optional(TileFormStore, {})
+      dashboardIsEdited: true,
+      imageUrl: types.maybe(types.string),
+      videoUrl: types.maybe(types.string)
     })
-    .actions((self) => ({}))
+    .actions((self) => ({
+      fetchDashboard: flow(function* fetchDashboard(spaceId: string) {
```
Comment (Daboo1998) [inline src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts:34]:
```suggestion
      fetchDashboard: flow(function* (spaceId: string) {
```
No need for extra name

File: src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts [hunk 12]
```diff
@@ -11,11 +22,97 @@ const DashboardStore = types.compose(
...
+                tile.permanentType === PermanentTypeEnum.MEME ||
+                tile.permanentType === PermanentTypeEnum.DESCRIPTION ||
+                tile.permanentType === PermanentTypeEnum.VIDEO)
+            ) {
+              self.dashboardIsEdited = false;
+            }
+          });
+        }
+      }),
+      updatePositions: flow(function* updatePositions(tiles: TileInterface[]) {
```
Comment (Daboo1998) [inline src/scenes/collaboration/stores/DashboardStore/DashboardStore.ts:56]:
```suggestion
      updatePositions: flow(function* (tiles: TileInterface[]) {
```

**Review by Daboo1998 — CHANGES_REQUESTED** [review]

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx (hunk 2 above)
Comment (aminbksi) [reply src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:39]: it's only for uploading image, no need for form, actually we cannot include form for images, like what we did in EventForm

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx (hunk 5 above)
Comment (aminbksi) [reply src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx:47]: it complains for return value, but after the api call I will return if it's Done, so donno what's the problem here it needs to be there

Comment (aminbksi) [reply src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx:47]: and the same for rest

Comment (Daboo1998) [reply src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/VideoTileForm/VideoTileForm.tsx:47]: `const isSucceed: boolean = await createTile(spaceId, data);`

File: src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx (hunk 4 above)
Comment (PAKovalev) [reply src/scenes/collaboration/pages/DashboardPage/components/templates/TileForm/components/ImageTileForm/ImageTileForm.tsx:119]: No. Don't need to use api stuff in components

**Review by PAKovalev — APPROVED** [review]

**Review by Daboo1998 — APPROVED** [review]

**Pull request merged**